frontend/internal-packages/agent/src/chat/workflow/nodes/generateDDLNode.ts

```diff
@@ -1,13 +1,66 @@
+import { QADDLGenerationAgent } from '../../../langchain/agents'
+import type { BasePromptVariables } from '../../../langchain/utils/types'
+import { convertSchemaToText } from '../../../utils/convertSchemaToText'
 import type { WorkflowState } from '../types'
 
+const NODE_NAME = 'generateDDLNode'
+
+interface PreparedDDLGeneration {
+  agent: QADDLGenerationAgent
+  schemaText: string
+}
+
+/**
+ * Prepare QA DDL generation
+ */
+async function prepareDDLGeneration(
+  state: WorkflowState,
+): Promise<PreparedDDLGeneration> {
+  const schemaText = convertSchemaToText(state.schemaData)
+  const agent = new QADDLGenerationAgent()
+
+  return {
+    agent,
+    schemaText,
+  }
+}
+
 /**
- * Generate DDL Node - Agent generates DDL
- * Performed by agent
+ * Generate DDL Node - QA Agent generates DDL
+ * Performed by qaAgent
+ *
+ * TODO: DDL generation using LLM is a temporary solution.
+ * In the future, DDL will be generated mechanically without LLM.
  */
 export async function generateDDLNode(
   state: WorkflowState,
 ): Promise<WorkflowState> {
-  return {
-    ...state,
+  try {
```

Comment:
(aside)(don't need to fix) I want to do try-catch by focusing on function calls that may throw an error.